### ISSUE TITLE
Hints breaks lines. `InvalidParams: Received a completion with overlapping edits`

### ISSUE THREAD

**Issue by ybnesm** [helix-editor/helix#5693]
### Summary

I know of an issue where you press `Enter` during a suggestion and some characters are removed. But this is slightly different issue (I believe).

If you can't reproduce it and don't understand what I'm talking about, let me know, Ill record a video later.

### Reproduction Steps

 

You don't need to create new project. The main point is to use rust and write `println` statement with `env!` macro.
I tried this:

1. `cargo new lolkek`
2. `cd lolkek; hx src/main.rs`
3. write it in the main: `println("{}", env!("CA`
4. Hit `Tab` any number of times

The preview is shown somewhere other than on this line. Also if you hit `Enter` it pastes selected hint in a strange place.

I expected this to happen:
Lsp and helix work the same as with other hints.

Instead, this happened:
Strange behavior.

I see this error only with the `env!` macro.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-01-26T21:39:07.798 helix_lsp::transport [ERROR] <- InvalidParams: Received a completion with overlapping edits, this is not LSP-compliant
2023-01-26T21:39:07.798 helix_view::editor [ERROR] editor error: Async job failed: protocol error: InvalidParams: Received a completion with overlapping edits, this is not LSP-compliant
```

</details>

### Platform

Pop!_OS 22.04 LTS

### Terminal Emulator

alacritty 0.11.0

### Helix Version

helix 22.12 (97083f88)

**Comment by the-mikedavis** [comment]
Are you setting `editor.auto-pairs` to `false` in the config?

Could you make an asciinema recording of this? And also provide the log while running in verbose mode (`hx -v`)? https://github.com/helix-editor/helix/wiki/FAQ#access-the-log-file

**Comment by ybnesm** [comment]
> Are you setting `editor.auto-pairs` to `false` in the config?

No, I just showed what I wrote, not what the editor pastes for me.

> And also provide the log while running in verbose mode (`hx -v`)

I've updated the log to the `Helix log` section above.

### Asciinema:
[![asciicast](https://asciinema.org/a/kTJNxfLbVA3ojGp5LO13PBwlt.png)](https://asciinema.org/a/kTJNxfLbVA3ojGp5LO13PBwlt)

**Comment by ybnesm** [comment]
Also, sometimes if the file is bigger than in the video, the `CARGO...` is even inserted on the wrong line and messes up my code.

**Comment by woojiq** [comment]
Fun fact: I can't reproduce this problem even with an old helix and old rust now (how did I have this problem before? :D ). There is just no completion when I type inside `env!`. Anyway, yes, it's fixed, thx.

**Comment by pascalkuthe** [comment]
I do get completions inside `env' but not when inside another macro (like `println`) so I guess RA disabled that at some point
